src/main/java/thewaypointers/trafficsimulator/simulation/models/vehicles/Car.java

```diff
@@ -77,7 +77,12 @@ public void calculateNextPosition(long timeStep, HashMap<Node, ArrayList<RoadEdg
                     VehicleManager.removeVehicle(currentRoad, this);
                     RoadEdge nextRoadEdge = calculateNextRoad(nodeGraphMap);
 
-                    currentRoad = nextRoadEdge.getRoad();
+                    try{
+                        currentRoad = nextRoadEdge.getRoad();
```

Comment:
Can you remove the try-catch? It will make stuff confusing if we get an exception here.